pandas/core/frame.py

```diff
@@ -8419,7 +8419,9 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
-        if numeric_only is not None and axis in [0, 1]:
+        if (
+            self._is_homogeneous_type and self._mgr.any_extension_types and axis == 0
+        ) or (numeric_only is not None and axis in [0, 1]):
```

Comment:
the axis in [0, 1] is superfluous

Reply:
test_any_all_np_func passes axis=None for reduction along both axes.